Add feature/rembg background removal feature

flacoonb opened the pull request:
<!-- Thank you for contributing! -->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/PhotoboothProject/photobooth/blob/dev/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "x" next to an item)

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Other, please explain:

#### What changes did you make? (Give an overview)

This pull request introduces a new AI-powered background removal feature for images, using the `rembg` library. It adds configuration options for enabling and customizing background removal, integrates the processing into the image workflow, and provides installation scripts. The changes are grouped below by theme.

---

### Feature: AI Background Removal Integration

- Added support for background removal using the `rembg` library in `applyEffects.php`, with configurable options for model selection, alpha matting, background image, and post-processing. The workflow detects and uses a Python virtual environment if available.
- **Reordered the rembg processing** to occur after image rotation but before other effects (like polaroid, frame, text) to ensure proper integration into the effect pipeline.
- Introduced a new Python script `rembg_processor.py` to handle the background removal and compositing with the selected background image.
- Added an installation script `install_rembg.sh` to automate the setup of the `rembg` library and its dependencies in a Python virtual environment.

### Configuration and UI

- Extended `configsetup.inc.php` to provide configuration options for enabling/disabling rembg, selecting the background image, choosing the AI model, and tuning advanced parameters such as alpha matting and image size.

### Code Quality and Standards

- Applied PHP CS Fixer (`npm run composer:cgl`) to fix indentation issues in `applyEffects.php`.
- Applied formatting (`npm run format`) to ensure consistent JS and SCSS code.

#### Is there anything you'd like reviewers to focus on?

Commit:
Add advanced rembg options with alpha-matting thresholds, erode size, and post-processing

- Add configuration options for rembg alpha-matting foreground/background thresholds and erode size
- Add post-processing option for rembg
- Update admin panel UI with new expert-level settings
- English translations for new options
- Update rembg processor script to handle new parameters
- Fix configuration schema to include default values for new options

Change-Id: I60d04948f3273c357e90c1fd317bfd2f16325c36

Commit:
Add install_rembg.sh script for automated rembg setup

- Create install_rembg.sh script to set up rembg in virtual environment
- Script checks for Python 3, creates venv, installs rembg and dependencies
- Should be run before enabling rembg feature in admin panel
- Includes verification of successful installation

Change-Id: I74a47d5a38c5ef90aaffa69b27a8d85928dce1ce

Commit:
Improve install_rembg.sh for cross-platform Linux compatibility

- Add support for Raspberry Pi OS, Debian, and Ubuntu (x64 and ARM)
- Auto-detect and handle sudo/root permissions
- Automatically install python3, python3-venv, and python3-pip if missing
- Support multiple package managers (apt, yum, dnf)
- Better error handling and user feedback
- Out-of-the-box compatibility for most Linux distributions

Change-Id: I5684d0b5a4dbbc74c4afba7ff043a3beac90ad61

Commit:
Add colors and status indicators to install_rembg.sh

- Add colored output for better user experience
- Implement status functions: success (green), error (red), warning (yellow), info (blue)
- Add step indicators with purple color and wrench emoji
- Include header banners for major sections
- Add spinner animation for long-running operations
- Improve verification output with individual success indicators
- Make the script more visually appealing and user-friendly

Change-Id: I0e531f80bbad25c109c84a0da7b903ca880ff989

andi34 commented on api/applyEffects.php:
```diff
@@ -159,6 +159,62 @@
+                            $logger->debug('Using rembg venv python3', ['path' => $venvPath]);
+                        } else {
+                            $pythonCmd = 'python3';
+                            $logger->warning('rembg venv not found, using system python3', ['venv_path' => $venvPath]);
```
$logger->info or $logger->debug or $logger->error

flacoonb replied:
Thank you.  
I've updated the logging to use `$logger->info`, `$logger->debug`, and `$logger->error` as suggested.

andi34 commented:
Missing strings
```
"pictures:picture_rembg_alpha_matting_foreground_threshold":"Foreground threshold",
"manual:pictures:picture_rembg_alpha_matting_foreground_threshold": "Alpha matting foreground threshold.",
"pictures:picture_rembg_alpha_matting_background_threshold": "Background threshold",
"manual:pictures:picture_rembg_alpha_matting_background_threshold": "Alpha matting background threshold.",
"pictures:picture_rembg_alpha_matting_erode_size": "Erosion size",
"manual:pictures:picture_rembg_alpha_matting_erode_size": "Alpha matting erosion size.",
"pictures:picture_rembg_post_processing": "Post process mask",
"manual:pictures:picture_rembg_post_processing": "Post process mask.",
```

Commit:
Add Magic Greenscreen admin panel section

- Move rembg settings from pictures to new magic_greenscreen section
- Add English translation for magic_greenscreen menu label
- Position magic_greenscreen section between custom and video in admin menu

Change-Id: I9791be1e2cd4cb1a969824e5dfe142d46234fd91

andi34 committed:
overhaul rembg config

Change-Id: Ie61646eeedaa2a8ed6004c0141a7dc08d8b13de0

Commit:
docs: Add Magic Greenscreen section to FAQ

Added comprehensive guide for AI background removal feature
Includes installation, configuration, usage, and troubleshooting
Follows existing FAQ format and structure

Change-Id: Ib0a17a165a1ef298c30fafcfed3e6756fb398943

andi34 committed:
src: make rembg own class to re-use

Change-Id: I2b7b44065916ddf3f842e8abe6d900b485f936dd

andi34 committed:
configsetup: use english language on rembg model selection

Change-Id: I59ee69a739058f7f3bdbfa2a25f6454af3356445

andi34 committed:
docs: update rembg requirements

Change-Id: Ibdffe2e21e6bcff28737528f139456c587879475

andi34 commented:
Updated requirements https://github.com/danielgatis/rembg?tab=readme-ov-file#requirements

Also not recommended Pi4 or older for optimal performance.

Possible integration into collage can be considered in future.

Please check if I missed anything, looks fine to me now with a good integration.
